### PR TITLE
github: port macOS sanity checks from travis

### PR DESCRIPTION
**Pull request by bboozzoo**
Now that a bunch of tests is running via github actions, port the macOS sanity checks too.

### CODE REVIEW

File: .github/workflows/macos-sanity.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.10.x"
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install squashfs from homebrew
+        uses: artemnovichkov/action-homebrew@master
+        args: "install squashfs"
```
Comment (bboozzoo) [inline .github/workflows/macos-sanity.yaml:20]: @zyga @stolowski should we do that manually instead? What's the right set of commands here?

Comment (zyga) [reply .github/workflows/macos-sanity.yaml:20]: I don't use homebrew but IIRC the commands are rather simple:
```
brew install squashfs
```

There's some nice documentation on https://brew.sh/index_pl

File: .github/workflows/macos-sanity.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: macOS sanity checks
+on:
+  # Only run on pull requests: not pushes
+  pull_request:
+    branches: [ "master", "release/**" ]
+
+jobs:
+  cla-check:
```
Comment (zyga) [inline .github/workflows/macos-sanity.yaml:8]: Probably not what you want

File: .github/workflows/macos-sanity.yaml (hunk 1 above)
Comment (stolowski) [reply .github/workflows/macos-sanity.yaml:20]: `brew install squashfs` pulls dependencies for squashfs: lz4, lzo and zstd; I'll try to verify if this is sufficient.

File: .github/workflows/macos-sanity.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: macOS sanity checks
```
Comment (zyga) [inline .github/workflows/macos-sanity.yaml:1]: I would also use something different here

File: .github/workflows/macos-sanity.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.10.x"
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install squashfs from homebrew
+        uses: artemnovichkov/action-homebrew@master
+        args: "install squashfs"
```
Comment (zyga) [inline .github/workflows/macos-sanity.yaml:20]: This line doesn't pass gh validation

File: .github/workflows/macos-sanity.yaml (hunk 3 above)
Comment (zyga) [reply .github/workflows/macos-sanity.yaml:1]: Bump to just say MacOS here

File: .github/workflows/macos-sanity.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Install squashfs from homebrew
+        run: |
+          brew install squashfs
+
+      - name: Install Go package dependencies
+        run: |
+          ./mkversion.sh
+          ./get-deps.sh
+          # extra dependency on darwin:
+          go get golang.org/x/sys/unix
```
Comment (zyga) [inline .github/workflows/macos-sanity.yaml:34]: Would be great to eventually teach get-deps about it

File: .github/workflows/macos-sanity.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          go build -o /tmp/snp github.com/snapcore/snapd/cmd/snap
+
+      - name: Runtime sanity checks
+        run: |
+          /tmp/snp download hello
+          /tmp/snp version
+          # TODO: homebrew appears to be broken, brew install of squashfs fails
+          # and goes unnoticed by travis
+          if command -v mksquashfs; then /tmp/snp pack tests/lib/snaps/test-snapd-tools/ /tmp ; fi
+        working-directory: ./src/github.com/snapcore/snapd
```
Comment (zyga) [inline .github/workflows/macos-sanity.yaml:48]: I would move it closer to name, before run

Comment (bboozzoo) [reply .github/workflows/macos-sanity.yaml]: done

File: .github/workflows/macos-sanity.yaml (hunk 5 above)
Comment (bboozzoo) [reply .github/workflows/macos-sanity.yaml:35]: done

Comment (anonymouse64) [reply .github/workflows/macos-sanity.yaml:35]: where did you do this?

File: .github/workflows/macos-sanity.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Build sanity checks
+        run: |
+          go build -o /tmp/snp github.com/snapcore/snapd/cmd/snap
+
+      - name: Runtime sanity checks
+        working-directory: ./src/github.com/snapcore/snapd
+        run: |
+          /tmp/snp download hello
+          /tmp/snp version
+          # TODO: homebrew appears to be broken, brew install of squashfs fails
```
Comment (stolowski) [inline .github/workflows/macos-sanity.yaml:46]: Weird, but I see it's preexisting comment. Would be interesting to know how it broke.

File: .github/workflows/macos-sanity.yaml (hunk 5 above)
Comment (bboozzoo) [reply .github/workflows/macos-sanity.yaml:35]: Sorry for the confusion. It's not done, I replied under the wrong comment.

**Pull request merged**